test/python/circuit/test_initializer.py

```diff
@@ -446,6 +447,64 @@ def test_global_phase_1q(self):
                 actual = zero & disentangler
                 self.assertEqual(target, actual)
 
+    def test_decompose_with_int(self):
+        """Test initialize with int arg decomposes to a StatePreparation and reset"""
+        qc = QuantumCircuit(2)
+        qc.initialize(2)
+        decom_circ = qc.decompose()
+        decom_circ2 = decom_circ.decompose()
+        dag1 = circuit_to_dag(decom_circ)
+        dag2 = circuit_to_dag(decom_circ2)
+
+        self.assertEqual(len(dag1.op_nodes()), 3)
+        self.assertEqual(dag1.op_nodes()[0].name, "reset")
+        self.assertEqual(dag1.op_nodes()[1].name, "reset")
+        self.assertEqual(dag1.op_nodes()[2].name, "state_preparation")
```

Comment:
Why is it necessary to convert to a `DAG` here?  Why not just compare based on `QuantumCircuit.data`?

Reply:
yeah i think it should work comparing `qc.data`, is there a particular reason not to do the check by converting to a dag? I was mainly just following the style we use for other decompose tests

Reply:
The only reason I mentioned it was because it looked odd to me, and I didn't immediately know the iteration order of `DAGCircuit.op_nodes` - there's no _required_ total ordering in a DAG (only a partial ordering, but you can make a total order by sequencing the wires, I think).